bin/reth/src/stage/run.rs

```diff
@@ -257,19 +257,20 @@ impl Command {
             checkpoint: Some(checkpoint.with_block_number(self.from)),
         };
 
-        while let ExecOutput { checkpoint: stage_progress, done: false } =
-            exec_stage.execute(&provider_rw, input).await?
-        {
-            input.checkpoint = Some(stage_progress);
+        loop {
+            poll_fn(|cx| exec_stage.poll_execute_ready(cx, input)).await?;
```

Comment:
we add an extension trait with a helper function that does this:

https://docs.rs/tower/latest/tower/trait.ServiceExt.html#method.ready

but perhaps not worth it, maybe in a followup